### PR TITLE
db_options: Expose remaining PlainTableFactoryOptions

### PR DESCRIPTION
**Pull request by glittershark**
This is intended to go along with
https://github.com/facebook/rocksdb/pull/11442.

### CODE REVIEW

File: src/db_options.rs [hunk 1]
```diff
@@ -3578,6 +3582,27 @@ pub enum ChecksumType {
     XXH3 = 4, // Supported since RocksDB 6.27
 }
 
+/// Used by [`PlainTableFactoryOptions`]
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
```
Comment (aleksuss) [inline src/db_options.rs:3587]: The rust version should be bumped. 1.60.0 doesn't support the `Default` trait for enums.

Comment (glittershark) [reply src/db_options.rs:3587]: That, or I could hand-impl the trait so we don't need an MSRV bump

File: src/db_options.rs [hunk 2]
```diff
@@ -3578,6 +3582,32 @@ pub enum ChecksumType {
     XXH3 = 4, // Supported since RocksDB 6.27
 }
 
+/// Used by [`PlainTableFactoryOptions`]
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EncodingType {
```
Comment (aleksuss) [inline src/db_options.rs:3588]: I guess the `EncodingType` is a too common name. I would rename it to `KeyEncodingType`.

**Comment by aleksuss** [comment]
Done in #815

**Pull request closed without merging**